src/python/test/test_alpha_complex.py

```diff
@@ -105,16 +107,18 @@ def test_filtered_alpha():
     else:
         assert False
 
-    assert simplex_tree.get_filtration() == [
-        ([0], 0.0),
-        ([1], 0.0),
-        ([2], 0.0),
-        ([3], 0.0),
-        ([0, 1], 0.25),
-        ([0, 2], 0.25),
-        ([1, 3], 0.25),
-        ([2, 3], 0.25),
-    ]
+    filtration_generator = simplex_tree.get_filtration()
+    assert(next(filtration_generator) == ([0], 0.0))
+    assert(next(filtration_generator) == ([1], 0.0))
+    assert(next(filtration_generator) == ([2], 0.0))
+    assert(next(filtration_generator) == ([3], 0.0))
+    assert(next(filtration_generator) == ([0, 1], 0.25))
+    assert(next(filtration_generator) == ([0, 2], 0.25))
+    assert(next(filtration_generator) == ([1, 3], 0.25))
+    assert(next(filtration_generator) == ([2, 3], 0.25))
+    with pytest.raises(StopIteration):
+        next(filtration_generator)
+
```

Comment:
https://stackoverflow.com/q/9983547/1918193 suggests several other ways to compare a generator with something. The simplest is `list(simplex_tree.get_filtration()) == ...`, and there are some fancier ones with izip_longest. What you did works, but it seems a bit painful to have to go through that everywhere.